src/resources/2022/AutumnAton.ts

```diff
@@ -0,0 +1,171 @@
+import {
+  availableAmount,
+  availableChoiceOptions,
+  handlingChoice,
+  Item,
+  Location,
+  runChoice,
+  toLocation,
+  totalTurnsPlayed,
+  visitUrl,
+  xpath,
+} from "kolmafia";
+import { get } from "../../property";
+
+export const item = Item.get("Autumn-Aton");
+
+/**
+ * Is the autumn-aton currently in your inventory, available to deploy?
+ */
+export function available(): boolean {
+  return availableAmount(item) > 0;
+}
+
+/**
+ * Do you own the autumn-aton?
+ */
+export function have(): boolean {
+  return get("hasAutumnaton") || available();
+}
+
+function checkLocations(html: string): Location[] {
+  return xpath(
+    html,
+    '//select[@name="heythereprogrammer"]//option[position()>1]/text()'
+  ).map((name) => toLocation(name));
+}
+
+const use = () => visitUrl("inv_use.php?pwd&whichitem=10954");
+
+/**
+ * @returns The current location the autumn-aton is questing in; null if it is not on a quest.
+ */
+export function currentlyIn(): Location | null {
+  return get("autumnatonQuestLocation");
+}
+
+/**
+ * Deploy the autumn-aton to a location of your choosing.
+ * @param target A location to send the autumn-aton to, or a prioritized list of locations to send it to, or a function to pick which location to send it to.
+ * @param upgrade Should we apply any upgrades we see available?
+ * @returns Where we ended up sending the autumn-aton; null if we didn't send it off.
+ */
+export function sendTo(
+  target: Location | Location[] | ((locations: Location[]) => Location),
+  upgrade = true
+): Location | null {
+  if (!available()) return null;
+
+  const pageHtml = use();
+
+  if (upgrade && availableChoiceOptions()[1]) runChoice(1);
+
+  const locationsAvailable = checkLocations(pageHtml);
+  const location =
+    target instanceof Location
+      ? target
+      : Array.isArray(target)
+      ? target.find((l) => locationsAvailable.includes(l))
+      : target(locationsAvailable);
+  if (!location) return null;
+  if (!locationsAvailable.includes(location)) return null;
+
+  runChoice(2, `heythereprogrammer=${location.id}`);
+  if (handlingChoice()) runChoice(3);
+  return currentlyIn();
```

Comment:
Why not return location? What do you do if Mafia fails to get the location for whatever reason?

Reply:
That's a good point; I was returning mafia's value because I want it to return falsy if it somehow fails. Which is more dangerous, a false negative or a false positive?

Reply:
You can keep going with a false negative as long as KoL received the query to send the robot. KoLMafia just might not know where you sent it or when it's returning.

In the event of a false positive, you can check if you have the autumn-aton again. So that's quickly recoverable.

Perhaps there's a safer option if you grab the result from `runChoice` though I'm not sure what the contents of that will be here.

Reply:
Excellent point; edited

I think we can grab the result from runChoice but that's presumably what mafia is doing as well, so I think the failure case would probably mess with the both of us.